modules/ding_nodelist/templates/ding_nodelist.ding_event.single.tpl.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * @file
+ * Ding event single item template.
+ */
+
+$items = $variables['item'];
+$item = new stdClass();
+if ($variables['conf']['sorting'] == 'event_date') {
+  // Get the object from the array in the case we are sorting by date.
+  foreach ($items as $k => $v) {
+    $item = $v;
+  }
+}
+$image_field = 'field_' . $item->type . '_list_image';
+$image = _ding_nodelist_get_dams_image_info($item, $image_field);
+$category = field_view_field('node', $item, 'field_ding_event_category', 'teaser');
+$event_date = _ding_nodelist_get_event_date($item);
+$event_date_formatted = _ding_nodelist_formated_ding_event_date($item);
+$library = field_view_field('node', $item, 'og_group_ref', 'default');
+$price = field_view_field('node', $item, 'field_ding_event_price', 'default');
+$fee = t('Free');
+if (is_array($price)) {
+  $fee = current($price);
+  $fee = $fee . ' ' . t('kr.');
+}
+$lead = field_get_items('node', $item, 'field_ding_event_lead');
+$teaser = field_get_items('node', $item, 'field_ding_event_body');
```

Comment:
I think the amount and complexity of PHP code in a template is getting out of hand here. Please consider moving this to a preprocess function.